hazelcast/src/main/java/com/hazelcast/nio/tcp/SocketReader.java

```diff
@@ -16,12 +16,62 @@
 
 package com.hazelcast.nio.tcp;
 
-import java.nio.ByteBuffer;
+import com.hazelcast.util.counters.Counter;
 
 /**
- * Reads content from a {@link ByteBuffer}.
+ * The SocketReader is responsible for reading data from the socket, on behalf of a connection, into a byte-buffer. Once the
+ * data is read into the ByteBuffer, this ByteBuffer is passed to the {@link ReadHandler} that takes care of the actual
+ * processing of the incoming data.
+ *
```

Comment:
I understand that this is the intent of this class, but there's nothing related to `socket, connection & byte-buffer` in this interface,  so it's kind of a bit confusing to me. Don't know where's better place to put it though.